Lesson16 (#18)
ReDI-School/js-berlin-2020-fall: lesson16.md

<!-- .slide: id="lesson16" -->

# Basic Frontend - Fall 2020

Lesson 16, Tuesday, 2020-10-20

---

### The DOM API

1. what's the DOM API?
  - it's an interface that lets us interact with our web page
  <!-- .element: class="fragment" -->

> 1. what can we do with?

Reviewer: ```suggestion
1. what can we do with it?
```

  - we can access HTML elements
  <!-- .element: class="fragment" -->
  - we can read from HTML elements
  <!-- .element: class="fragment" -->

>   - we can change HTML elements content

Reviewer: ```suggestion
  - we can change HTML elements' content
```

Changing the value of an `input`:
```html
<input type="text" id="color" value="green">
```

```js
let colorElement = document.getElementById('color');
colorElement.value = "red";
let color = colorElement.value;
console.log(color); // red
```
<!-- .element: class="fragment" -->

---

### The DOM API: examples

Changing the style of a `div`:
```html
<div id="content">Hello</div>
```

```js
let contentElement = document.getElementById('content');
contentElement.style.backgroundColor = "black";
contentElement.style.color = "white";
contentElement.style.fontSize = "32px";
contentElement.style.padding = "16px 32px";
```
<!-- .element: class="fragment" -->

---

### The DOM API: examples

Changing the content of a `div`:

```html
<div id="content">Hello</div>
```

```js
let contentElement = document.getElementById('content');
contentElement.textContent = "Hi!";
```
<!-- .element: class="fragment" -->

---

### The DOM API: examples

Changing the whole `body` element:

```js
document.body.style.backgroundColor = "black";
document.body.style.color = "white";
document.body.style.fontSize = "48px";
document.body.textContent = "Hello!";
```

---

### The DOM API: examples

> Clear the content of a `div`, a `paragraph` and an `input`:

Reviewer: ```suggestion
Clear the contents of a `div`, a `paragraph` and an `input`:
```

```html
<div id="myDiv">I'm the content of a div</div>
<p id="myParagraph">I'm a paragrpah</p>
<input type="text" id="myInput" value="bla">
```

```js
let myDivElement = document.getElementById('myDiv');
let myParagraphElement = document.getElementById('myParagraph');
let myInputElement = document.getElementById('myInput');

myDivElement.textContent = '';
myParagraphElement.textContent = '';
myInputElement.value = '';
```
<!-- .element: class="fragment" -->

---

### The DOM API

> What's the difference between those element: `h1`, `h2`, `p`, `div`

Reviewer: ```suggestion
What's the difference between those elements: `h1`, `h2`, `p`, `div`
```

```html
<h1>This is the main heading</h1>
<h2>This is the sub heading</h2>
<p>And here is a nice <span>paragraph</span> with a span inside.</p>
<div><a href="http://example.com">A link</a> inside a div.</div>
```

and those: `input`, `img`
```html
<input type="text" id="color" value="red">
<img src="myImage.jpg">
```

> The content of `input` and `img` is controlled by an element `attribute`.

Reviewer: ```suggestion
The contents of `input` and `img` are controlled by an element' `attribute`.
```
